### ISSUE TITLE
Bash history import issues

### ISSUE THREAD

**Issue by cyqsimon** [ellie/atuin#745]
I spotted two issues while trying to import my Bash history:

1. Import ordering is incorrectly reversed.
2. `.bash_history` can be configured to store timestamps, a case not handled.

---

## Ordering

In the following snippet, lines from the top of the file will receive the newest timestamp:

https://github.com/cyqsimon/atuin/blob/d46e3ad47d03e53232269973806fb9da2248ba19/atuin-client/src/import/bash.rs#L42-L67

This is incorrect for `.bash_history` - new items are appended to the end of the file, not the start. So the first lines are in fact the oldest commands.

## Stored timestamps

The current import code for Bash assumes that each line is a command. However if `HISTTIMEFORMAT` is set, then a timestamp is recorded for each command (see `man bash`). So `.bash_history` would look something like this:

```
#1672918999
git reset
#1672919006
git clean -dxf
#1672919020
cd ../
```

This case is currently not handled, so the timestamp lines are getting falsely-imported as commands.

---

I am working on a patch that will fix both these issues, so give me a little while and I'll submit a PR.

The reason I'm posing this issue mostly concerns the extent of the ordering problem. Did it happen because
- the original author used a different shell that does store the newest commands at the start of its history file, and wrongly assumed the same for Bash?
- or the original author made a simple arithmetic mistake, and this is an issue for the import code of all shells?

In other words, I'm not that familiar with other shells, so does the import code for them need fixing as well?

**Comment by ellie** [comment]
> In other words, I'm not that familiar with other shells, so does the import code for them need fixing as well?

I don't believe so - initially we only supported ZSH, as that's what I use + have tested extensively with a whole bunch of my machines. I never really had any bash history to test with, and I don't think @conradludgate has either. 

Our Fish/resh/histdb code has had more eyes on it too, and they use a more structured way of storing history
